Adding script for generating table of parameters

matt-graham opened the pull request:
Work in progress but should resolve #1419 

Script can be run from root of respository with

```
python docs/tlo_parameters.py resources parameters.md
```

with this generating a Markdown output `parameters.md` which can be converted to HTML with

```
pandoc -s parameters.md -o parameters.html --css pandoc.css --self-contained
```

using CSS from https://gist.github.com/killercup/5917178

I'll hopefully be able to get this to also output reStructuredText to allow running as part of the documentation build and including on the website.

matt-graham commented on src/tlo/methods/healthsystem.py:
```diff
@@ -165,7 +165,7 @@ class HealthSystem(Module):
             Types.STRING, "If `actual`, then use the numbers and distribution of staff estimated to be available"
                           " currently; If `funded`, then use the numbers and distribution of staff that are "
-                          "potentially available. If 'funded_plus`, then use a dataset in which the allocation of "
+                          "potentially available. If `funded_plus`, then use a dataset in which the allocation of "
```
Change here is to prevent Markdown rendering in parameter table getting disrupted by non-matching quotation marks

tbhallett commented:
I've made an edit in [this](8b315ded68f4bd4bcdaa8f1c6a152f1cfb696886) commit (floating loose in the repo) to make the pagination more manageable (by excluding two parameters, which are really just datasets). Also, excluded the `Wasting` module that has no parameters.

matt-graham commented:
Thanks @tbhallett - I've applied the patch you suggested and also now updated to allow outputting a reStructuredText file as well as Markdown to allow inclusion in our docs, and updated the docs build to automatically generate this page.

tbhallett commented:
Thanks @matt-graham  -- that's brilliant and very timely. It will be great to have this on the website.

matt-graham commented:
@tbhallett just to check are you happy for this to be merged in? Know that you already approved but as I made a few subsequent changes after that wanted to make sure!

tbhallett commented:
Yes please do merge in. Thanks again. And, if you can, please could you also regenerate the website so that it appears there?
